contents/code/tilelist.js

```diff
@@ -251,6 +251,24 @@ TileList.prototype.addClient = function(client) {
         return;
     }
 
+    if (client.setMaximize != null) {
+        // Unmaximize the new client.
+        client.setMaximize(false, false);
+
+        // Unmaximize the active client if there is already maximized one
+        // on the same desktop.
+        var screen = client.screen;
+        var desktop = client.desktop;
+        this.tiles.forEach(function(t) {
+            if (t.maximized == true &&
+                t._currentDesktop == desktop && t._currentScreen == screen) {
```

Comment:
I'm accessing private members here. I'm aware of it.

Reply:
I'd like it if you made them non-private or added a proxy "isOn(desktop, screen)" method or something.

Though TBH it's not a death sentence.

Reply:
Yes, I already have `Tile.prototype.getDesktop()` in #137.

Reply:
I consider this one to be just a WIP, since there's no support in stable upstream anyway.

Reply:
OK, the private members are no longer accessed directly.